tests/test_transformations.py

```diff
@@ -1959,84 +1965,949 @@ def test_delete_coordinate_system_exceptions(
+            constructor. For each list entry a CSM instance is generated
+        cs_data :
+            A list containing the data to create coordinate systems. Each entry is a
+            tuple containing the list index of the target CSM instance and the
+            arguments that should be passed to the ``create_cs`` method
+        merge_data :
+            A list of tuples. Each tuple consists of two indices. The first one is the
+            index of the source CSM and the second one of the target CSM. If an entry
+            is `None`, it is skipped and no merge operation is performed
+        csm_diffs :
+            A list of modifications that should be applied to the ``csm_data`` before
+            creating the second set of CSM instances. Each entry is a tuple containing
+            the index and new value of the data that should be modified.
+        cs_diffs :
+            A list of modifications that should be applied to the ``cs_data`` before
+            creating the coordinate systems of the second set of CSM instances. Each
+            entry is a tuple containing the index and new value of the data that should
+            be modified.
+        merge_diffs :
+            A list of modifications that should be applied to the ``merge_data`` before
+            merging the second set of CSM instances. Each entry is a tuple containing
+            the index and new value of the data that should be modified.
+        exp_results :
+            A list containing the expected results of each instance comparison
+
+        """
+        # define support function
+        def create_csm_list(csm_data_list, cs_data_list, merge_data_list):
+            """Create a list of CSM instances."""
+            csm_list = []
+            for args in csm_data_list:
+                csm_list += [tf.CoordinateSystemManager(*args)]
+
+            for data in cs_data_list:
+                csm_list[data[0]].create_cs(*data[1])
+
+            for merge in merge_data_list:
+                if merge is not None:
+                    csm_list[merge[1]].merge(csm_list[merge[0]])
+
+            return csm_list
+
+        # create diff inputs
+        csm_data_diff = deepcopy(csm_data)
+        for diff in csm_diffs:
+            csm_data_diff[diff[0]] = diff[1]
+
+        cs_data_diff = deepcopy(cs_data)
+        for diff in cs_diffs:
+            cs_data_diff[diff[0]] = diff[1]
+
+        merge_data_diff = deepcopy(merge_data)
+        for diff in merge_diffs:
+            merge_data_diff[diff[0]] = diff[1]
+
+        # create CSM instances
+        csm_list_1 = create_csm_list(csm_data, cs_data, merge_data)
+        csm_list_2 = create_csm_list(csm_data_diff, cs_data_diff, merge_data_diff)
+
+        # test
+        for i, _ in enumerate(csm_list_1):
+            assert (csm_list_1[i] == csm_list_2[i]) is exp_results[i]
+            assert (csm_list_1[i] != csm_list_2[i]) is not exp_results[i]
```

Comment:
do we need the second assert? looks to do the same? (`==` = `not !=`?)

Reply:
Well, technically, one can implement a separate function for `!=` and it doesn't hurt to double-check I guess